### PR TITLE
Fix array plus one v3

### LINKED ISSUE #431: Test fail. arrays/plus_one makes tests fail
The function plus_one_v3() tries to access the index as if it was an actual index (number) but since it's being used in a for loop with an enumerate, it is actually a tuple of (idx,value) for all the items in the list

I discovered this while making a PR for another issue since it would reject my build.

Already making a PR changing it so the tests run again.

### PR DESCRIPTION
**Pull request by krimvp**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [ ] **If creating a new file :**
  - [ ] added links to it in the README files ?
  - [ ] included tests with it ?
  - [ ] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [X] **if done some changes :**
  - [X] wrote short description in the PR explaining what the changes do ?
  - [X] Fixes #431 ad #337 

- [ ] **other**

### CODE REVIEW

File: algorithms/linkedlist/copy_random_pointer.py [hunk 1]
```diff
@@ -4,35 +4,90 @@
...
+
+
+class RandomListNode(object):
+    def __init__(self, label):
+        self.label = label
+        self.next = None
+        self.random = None
+
+
+class Solution0:
```
Comment (keon) [inline algorithms/linkedlist/copy_random_pointer.py]: you can unwrap this class and just leave it as a function

File: algorithms/linkedlist/copy_random_pointer.py [hunk 2]
```diff
@@ -4,35 +4,90 @@
...
-  #   copy = defaultdict(lambda: RandomListNode(0))
-  #   copy[None] = None
-  #   node = head
-  #   while node:
-  #       copy[node].label = node.label
-  #       copy[node].next = copy[node.next]
-  #       copy[node].random = copy[node.random]
-  #       node = node.next
-  #   return copy[head]
+class Solution1:  # O(n)
```
Comment (keon) [inline algorithms/linkedlist/copy_random_pointer.py]: ditto

File: algorithms/linkedlist/copy_random_pointer.py [hunk 3]
```diff
@@ -4,35 +4,90 @@
...
+        node = head
+        while node:
+            copy[node].label = node.label
+            copy[node].next = copy[node.next]
+            copy[node].random = copy[node.random]
+            node = node.next
+        return copy[head]
+
+
+class TestSuite(TestCase):
```
Comment (keon) [inline algorithms/linkedlist/copy_random_pointer.py]: could you move this to [tests folder](https://github.com/keon/algorithms/blob/master/tests/test_linkedlist.py)?

File: algorithms/linkedlist/__init__.py [hunk 4]
```diff
@@ -6,3 +6,4 @@
 from .is_cyclic import *
 from .merge_two_list import *
 from .is_palindrome import *
+from .copy_random_pointer import *
```
[the remark this answers is not in the document]
Comment (krimvp) [reply algorithms/linkedlist/__init__.py]: Oops my fault. Should be fine now

**Pull request merged**